File: solutions/vbnet/darts/1/Darts.vb
Imports System

Public Module Darts
    Public Function Score(ByVal x As Double, ByVal y As Double) As Integer
        Select Case Distance(x, y)
            Case <= 1.0
                Return 10
            Case <= 5.0
                Return 5
            Case <= 10.0
                Return 1
            Case Else
                Return 0
        End Select
    End Function

    Private Function Distance(ByVal x As Double, ByVal y As Double) As Double
        Return Math.Sqrt(x * x + y * y)
    End Function
End Module

File: solutions/delphi/darts/1/darts.pas
unit uDarts;

interface

type
  TDarts = class
    class function Score(const x,y:extended):Integer;
  end;

implementation

uses
  System.Sysutils, System.Math;

{ TDarts }

class function TDarts.Score(const x, y: extended): Integer;
var
  h:Extended;
begin
  Result := 0;
  h := sqrt(Power(x, 2) + Power(y, 2));
  if (h <= 10) then
    Result := 1;
  if (h <= 5) then
    Result := 5;
  if (h <= 1) then
    Result := 10;
end;

end.